terraform-modules/aws/nlb/main.tf

```diff
@@ -0,0 +1,27 @@
+resource "aws_lb" "nlb" {
+  name               = var.nlb_name
+  internal           = var.enable_internal 
+  load_balancer_type = "network"
+  subnets            = var.nlb_subnets
+
+  enable_deletion_protection = var.enable_deletion_protection
+
+dynamic "access_logs" {
+    # The contents of the list is irrelevant. The only important thing is whether or not to create this block.
+    for_each = var.enable_nlb_access_logs
```

Comment:
Just making sure.  This is optional?  If `var.enable_nlb_access_logs` is an empty list this module would still work?

Reply:
no it won't work with an empty list.

Reply:
Im not fully understanding.  How is it optional but fails when a user dont add in an input or dont want to output the logs?

Reply:
it is optional when a user don't add inputs. when user add inputs it shouldn't be an empty list

Reply:
Can you make it crystal clear for me and also document this in the description of this input.

If i dont want to output the NLB logs to S3, what do I put in as this input?  

If i do want to output the NLB logs to S3, what do I put in as this input?

Reply:
This is Optional, Only used to output the NLB logs to S3. we need to input bucket_name and bucket_prefix to the list. if left empty it will not output the logs to S3

updated the description